lib/datadog/appsec/configuration/settings.rb

```diff
@@ -53,73 +52,71 @@ def self.add_settings!(base)
               end
 
               option :ruleset do |o|
-                o.default { ENV.fetch('DD_APPSEC_RULES', DEFAULT_APPSEC_RULESET) }
+                o.env_var 'DD_APPSEC_RULES'
+                o.default DEFAULT_APPSEC_RULESET
               end
 
               option :ip_denylist do |o|
-                o.default { [] }
+                o.type :array
+                o.default []
               end
 
               option :user_id_denylist do |o|
-                o.default { [] }
+                o.type :array
+                o.default []
               end
 
               option :waf_timeout do |o|
-                o.default { ENV.fetch('DD_APPSEC_WAF_TIMEOUT', DEFAULT_APPSEC_WAF_TIMEOUT) } # us
```

Comment:
We probably want to capture the units (`us`), even if just as a comment like before. Same for the `trace_rate_limit` option below.